packages/client/pages/index.tsx

```diff
@@ -37,11 +39,27 @@ const Home: NextPage = () => {
         `Finding deals within ${MAX_DISTANCE_IN_METERS}m of ${userLatitude}, ${userLongitude}`
       )
       const data = await api.findDeals(userLatitude, userLongitude)
-      console.log(data)
       setCurrentDeals(data.deals)
+      populateCategories(data.deals)
     })()
   }, [isLocationValid, userLatitude, userLongitude])
 
+  const populateCategories = (deals: Deal[]) => {
+    const categories = new Set(
+      deals.map((deal) => deal.merchantOutlet.merchant.category)
+    )
+    const categoryMap: Map<string, boolean> = new Map()
+    categories.forEach((category) => categoryMap.set(category, true))
+    setCategories(categoryMap)
+  }
+
+  const handleButtonClick = (item: string) => {
+    const newMap = new Map(categories)
+    const oldValue = newMap.get(item)
+    newMap.set(item, !oldValue)
+    setCategories(newMap)
```

Comment:
```suggestion
    setCategories(categories => {
      const newMap = new Map(categories)
      const oldValue = newMap.get(item)
      newMap.set(item, !oldValue)
      return newMap
    })
```
minor performance change